### PR TITLE
Update the appearance of the "Add a domain" button

### PR DESCRIPTION
**Pull request by rafaelgallani**
#### Changes proposed in this Pull Request
We want to update the appearance of the "Add a domain" button at the top of the Site Domains page, as specified in: 1146722293412575-as-1200601027531795.

#### Testing instructions
Use the "Link to Calypso live" link below to view the build for this branch.
Go to the site domains page (My Sites -> Manage -> Domains).
Make sure that the button looks like the "New" version shown in the screenshot below:

Old button:
image

New button:
image

### CODE REVIEW

File: client/my-sites/domains/domain-management/list/options-domain-button.jsx
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/list/options-domain-button.jsx:71]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'Manage all your domains' )` *ES Score: 9*

**Comment by rafaelgallani** [comment]
Oops. Looks like some E2E tests need a few updates. I'm adding them... 😸

**Review by delputnam — CHANGES_REQUESTED** [review]
The changes look good, could you please also disable the failing tests for now.  We'll add them back in with the PR for the new "Add a domain" button.

**Review by delputnam — APPROVED** [review]

**Comment by delputnam** [comment]
This LGTM now.

Noting here our task to update and re-enable the disabled tests: 1146722293412575-as-1200616251991847

Please wait for a review from either @kelliepeterson  or @poligilad-auto re: design correctness before deploying.

**Review by poligilad-auto — COMMENTED** [review]
Very cool. I'm not sure how the component works, but can we have it align like this to the edge of the button?

image

I also saw it aligned similarly in 'Media' (or is it just aligning to the center of the content?)
image

**Comment by delputnam** [comment]
@rafaelgalani I did notice one other style change we need to make. For the mobile-sized breakpoint, we'll want to add a margin to this button so that it's not flush with the right side of the screen. (You can also update the deprecated breakpoint mixin to use the new `break-mobile` at the same time, I guess.

Screen Shot 2021-07-16 at 2 35 03 PM

**Comment by delputnam** [comment]
> ...can we have it align like this to the edge of the button?

Yep. That's possible using the `customPosition` prop.

**Comment by rafaelgallani** [comment]
> I also saw it aligned similarly in 'Media' (or is it just aligning to the center of the content?)

@poligilad-auto Yes, it is aligning with the center of the content. In this case, the media button. I tried doing similarly without adding too much custom code. Can you take a look again and see if that's better?

I also added the mobile breakpoints:

image

**Comment by worldomonation** [comment]
I see that two steps are skipped as part of this. I see that you were able to get the Desktop E2E test to pass by skipping the step, but it seems in subsequent pushes most of the tests are no longer running.

For the `Manage Domains - Add a Domain: Add a Domain` step, the steps to fix the test should be:
- modify [this line](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/pages/domains-page.js#L11) to the new CSS selector.
- potentially remove [this line](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/specs-calypso/wp-manage-domains__add-a-domain-spec.js#L48) if it is no longer needed.

The same thing should be done for `Manage Domains - Map a Domain: Add a domain` step.

**Review by worldomonation — CHANGES_REQUESTED** [review]
Requesting changes and e2e to be enabled.

**Comment by rafaelgallani** [comment]
> * potentially remove [this line](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/specs-calypso/wp-manage-domains__add-a-domain-spec.js#L48) if it is no longer needed.

@worldomonation
From what I could understand, apparently, each test of these E2E specs is dependent on the previous ones. 
So, if I remove the line you mentioned, all the other ones would fail - as they did on this last run. Since each run depends on the previous one, if we remove the side-effect from the first one, all the next ones will fail.

We were planning to remove these tests temporarily (as mentioned in 1146722293412575-as-1200616251991847), hence the `.skip` usage, but because of this dependency, then they will fail either way.
So, for now, I will just remove the tests, and later we will add them again, after all changes have been deployed (1146722293412575-as-1200616251991847).

**Review by worldomonation — APPROVED** [review]

**Comment by worldomonation** [comment]
> We were planning to remove these tests temporarily (as mentioned in 1146722293412575-as-1200616251991847), hence the `.skip` usage, but because of this dependency, then they will fail either way.
> So, for now, I will just remove the tests, and later we will add them again, after all changes have been deployed (1146722293412575-as-1200616251991847).

Sure, as long as there is a plan to re-enable the disabled tests, then it's OK from the Quality Squad's perspective.

Please feel free to reach out to us (slack #quality-squad) if you need help on this.

File: client/my-sites/domains/domain-management/list/add-domain-button.jsx
Comment (delputnam) [inline client/my-sites/domains/domain-management/list/add-domain-button.jsx]: I know that you're not using this file in this PR, but it will be needed for #54649, so let's keep it even though it's not used in the context of this PR.

**Review by delputnam — APPROVED** [review]
One final request to not delete `add-domain-button.jsx` and I think we're ready to merge pending design review.

**Comment by poligilad-auto** [comment]
That better @rafaelgalani, thanks!
I think it's fine.

In a perfect world, I would say it's best to align the edge of the popover menu to the edge of the button + align the arrow to the middle of the button. But I don't want to over complicate things.

Some weird behaviour I did see is that the mobile version of the popover menu is narrower. If possible would be good to keep it to the same width.

**Comment by rafaelgallani** [comment]
@poligilad-auto 
I'm adding the changes that we talked about on Friday: 
- Centralize popover to the button;
- Make sure popover is well-fitted in the mobile viewport.

Could you take a final look and see if that's ok? 😸 
Also, here's how it looks on mobile (using Chrome emulator, iPhone X, 100%):

image

**Comment by poligilad-auto** [comment]
Looks great @rafaelgalani!

**Review by poligilad-auto — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6322985

Thank you @rafaelgalani for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6322985).

**Pull request merged**
